bin/hera/src/hera.rs

```diff
@@ -140,8 +141,35 @@ impl<Node: FullNodeComponents> HeraExEx<Node> {
         Self { ctx, cfg }
     }
 
-    /// Starts the Execution Extension loop.
+    /// Wait for the L2 genesis L1 block (aka "origin block") to be available in the L1 chain.
+    async fn wait_for_l2_genesis_l1_block(&mut self) -> Result<()> {
+        loop {
+            if let Some(notification) = self.ctx.notifications.recv().await {
+                if let Some(committed_chain) = notification.committed_chain() {
+                    let tip = committed_chain.tip().block.header().number;
+                    // TODO: commit the chain to a local buffered provider
+                    // self.chain_provider.commit_chain(committed_chain);
+
+                    if let Err(err) = self.ctx.events.send(ExExEvent::FinishedHeight(tip)) {
+                        bail!("Critical: Failed to send ExEx event: {:?}", err);
+                    }
+
+                    if tip >= self.cfg.genesis.l1.number {
+                        break Ok(());
+                    } else {
+                        debug!(target: "hera", "Chain not yet synced to rollup genesis. L1 block number: {}", tip);
+                    }
+                }
+            }
+        }
+    }
+
+    /// Starts the Hera Execution Extension loop.
     pub async fn start(mut self) -> Result<()> {
-        unimplemented!()
+        // Step 1: Wait for the L2 origin block to be available
+        self.wait_for_l2_genesis_l1_block().await?;
+        info!(target: "hera", "Chain synced to rollup genesis");
```

Comment:
ExEx will already have a tracing span with the same name as the ExEx itself, so we can remove `target` here